textattack/goal_functions/classification/input_reduction.py

```diff
@@ -0,0 +1,36 @@
+from .classification_goal_function import ClassificationGoalFunction
+
+
+class InputReduction(ClassificationGoalFunction):
+    """
+    An targeted attack on classification models which attempts to maximize the 
+    score of the target label until it is the predicted label.
+    """
+
+    def __init__(self, *args, target_num_words=1, **kwargs):
+        self.target_num_words = target_num_words
+        super().__init__(*args, **kwargs)
+
+    def _is_goal_complete(self, model_output, attacked_text):
+        return (
+            self.ground_truth_output == model_output.argmax()
+            and attacked_text.num_words <= target_num_words
+        )
+
+    def _should_skip(self, model_output, attacked_text):
+        return self.ground_truth_output != model_output.argmax()
+
+    def _get_score(self, model_output, attacked_text):
+        if self.ground_truth_output != model_output.argmax():
+            return float("-inf")
+        cur_num_words = attacked_text.num_words
+        initial_num_words = self.initial_attacked_text.num_words
+        num_words_score = (initial_num_words - cur_num_words) / initial_num_words
+        model_score = model_output[self.ground_truth_output]
+        return num_words_score + model_score / initial_num_words
```

Comment:
1. can you add a comment explaining why this score is a sum of two things (model score is used as a tiebreaker)

2. this can overflow above 1, right? so maybe either [1] clamp at 1, that should be fine or [2] divide by (initial_results + 1) -- I think that would look like ```return  (initial_num_words - cur_num_words + model_score) / (initial_num_words + 1)```